app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -46,10 +46,17 @@ def self.raw_connect(hostname, port, token, alerts = false)
     end
 
     def connect(options = {})
-      @client ||= self.class.raw_connect(hostname,
-                                         port,
-                                         authentication_token('default'),
-                                         options[:alerts])
+      @clients ||= {}
+      @clients[options[:alerts] ? :alerts : :metrics] ||= self.class.raw_connect(
```

Comment:
I don't like this `options[:alerts] ? :alerts : :metrics` inside the hash accessor, can you extract it into a variable? Would make a lot more sense.